_sources/Conditionals/Nestedconditionals.rst

```diff
@@ -0,0 +1,104 @@
+..  Copyright (C)  Brad Miller, David Ranum, Jeffrey Elkner, Peter Wentworth, Allen B. Downey, Chris
+    Meyers, and Dario Mitchell.  Permission is granted to copy, distribute
+    and/or modify this document under the terms of the GNU Free Documentation
+    License, Version 1.3 or any later version published by the Free Software
+    Foundation; with Invariant Sections being Forward, Prefaces, and
+    Contributor List, no Front-Cover Texts, and no Back-Cover Texts.  A copy of
+    the license is included in the section entitled "GNU Free Documentation
+    License".
+
+.. qnum::
+   :prefix: condition-8-
+   :start: 1
+
+.. index::
+   single: nested conditionals
+   single: conditional; nested
+
+Nested conditionals
+-------------------
+
+One conditional can also be **nested** within another. For example, assume we have two integer variables, ``x`` and ``y``.
+The following pattern of selection shows how we might decide how they are related to each other.
+
+.. sourcecode:: python
+
+    if x < y:
+        print("x is less than y")
+    else:
+        if x > y:
+            print("x is greater than y")
+        else:
+            print("x and y must be equal")
+
+The outer conditional contains two branches.
+The second branch (the else from the outer) contains another ``if`` statement, which
+has two branches of its own. Those two branches could contain
+conditional statements as well.
+
+The flow of control for this example can be seen in this flowchart illustration.
+
+.. image:: Figures/flowchart_nested_conditional.png
+
+
+
+
+Here is a complete program that defines values for ``x`` and ``y``.  Run the program and see the result.  Then change the values of the variables to change the flow of control.
+
+.. activecode:: ac7_8_1
+
+    x = 10
+    y = 10
+
+    if x < y:
+        print("x is less than y")
+    else:
+        if x > y:
+            print("x is greater than y")
+        else:
+            print("x and y must be equal")
+
+.. note::
+
+    In some programming languages, matching the if and the else is a problem. However, in Python this is not 
+    the case. The indentation pattern tells us exactly which else belongs to which if.
+
+If you are still a bit unsure, here is the same selection as part of a codelens example.  Step through it to see how the correct ``print`` is chosen.
+
+.. codelens:: clens7_8_1
```

Comment:
Something is funny here, after the first line I lose the red arrow...

Reply:
Huh, that doesn't happen when I run the page for some reason.

Reply:
OK, I'll look into it more on my end.

Reply:
I think I'm using the github version of runestone - I don't recall using pypi.

Reply:
If you did ``pip install runestone`` then you are using the released pypi version.  If you did ``pip install -e /path/to/components`` . Then you are using github...  If you do pip list and you see runestone version 3.0.9 with a path next to it that also shows you are using github.  If no path then standard release.

Reply:
I'm seeing version 3.0.7 with a path!

Reply:
Then I think you haven't done a git pull into that path for a while!

Reply:
When I have a CodeLens it shows up like this:

<img width="841" alt="screen shot 2018-06-07 at 1 13 08 pm" src="https://user-images.githubusercontent.com/51115/41119431-62c875a8-6a58-11e8-9dd9-6be6a8aadd91.png">

Notice that the last button is below, this does not happen on other versions.  Are you seeing the same?

Reply:
Nope, in my view the last button is in line with the rest of the buttons.

Reply:
Wow, crazy.  Could you tar and gzip your source fopp folder and email it to me?  I want to check to see if I can find something different between them.  I've tried a clean check and a build with the released version of runestone and I still keep seeing this weird problem.